common/exchange_input.go

```diff
@@ -104,20 +105,22 @@ type ServiceChoiceFile struct {
 
 // This is used when reading json file the user gives us as input to create the service
 type ServiceFile struct {
-	Org                 string                       `json:"org"` // optional
-	Label               string                       `json:"label"`
-	Description         string                       `json:"description"`
-	Public              bool                         `json:"public"`
-	Documentation       string                       `json:"documentation"`
-	URL                 string                       `json:"url"`
-	Version             string                       `json:"version"`
-	Arch                string                       `json:"arch"`
-	Sharable            string                       `json:"sharable"`
-	MatchHardware       map[string]interface{}       `json:"matchHardware,omitempty"`
-	RequiredServices    []exchange.ServiceDependency `json:"requiredServices"`
-	UserInputs          []exchange.UserInput         `json:"userInput"`
-	Deployment          interface{}                  `json:"deployment"` // interface{} because pre-signed services can be stringified json
-	DeploymentSignature string                       `json:"deploymentSignature,omitempty"`
+	Org                  string                       `json:"org"` // optional
+	Label                string                       `json:"label"`
+	Description          string                       `json:"description"`
+	Public               bool                         `json:"public"`
+	Documentation        string                       `json:"documentation"`
+	URL                  string                       `json:"url"`
+	Version              string                       `json:"version"`
+	Arch                 string                       `json:"arch"`
+	Sharable             string                       `json:"sharable"`
+	MatchHardware        map[string]interface{}       `json:"matchHardware,omitempty"`
+	RequiredServices     []exchange.ServiceDependency `json:"requiredServices"`
+	UserInputs           []exchange.UserInput         `json:"userInput"`
+	Deployment           interface{}                  `json:"deployment"` // interface{} because pre-signed services can be stringified json
+	DeploymentSignature  string                       `json:"deploymentSignature,omitempty"`
+	ClusterDeployment    string                       `json:"clusterDeployment"`
```

Comment:
This needs to be an interface{} not a string

Reply:
Fixed in the new code.